src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DynamicInterfaceCastableImplementation.Fixer.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices
+{
+    public abstract class DynamicInterfaceCastableImplementationFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds =>
+            ImmutableArray.Create(
+                DynamicInterfaceCastableImplementationAnalyzer.InterfaceMembersMissingImplementationRuleId,
+                DynamicInterfaceCastableImplementationAnalyzer.MeembersDeclaredOnImplementationTypeMustBeSealedRuleId);
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxGenerator generator = SyntaxGenerator.GetGenerator(context.Document);
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode enclosingNode = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            SyntaxNode declaration = generator.GetDeclaration(enclosingNode);
+            if (declaration == null || !CodeFixSupportsDeclaration(declaration))
+            {
+                return;
+            }
+            Diagnostic diagnostic = context.Diagnostics.First();
```

Comment:
💡 This should `foreach` over `context.Diagnostics`

Reply:
@sharwell fwiw, I've seen fixers with the 3 patterns: `.Diagnostics.First()`, `.Diagnostics[0]`, and `foreach` across the repository.

So if the `foreach` is always the recommended approach, this could be cleaned up in couple other fixers.